resolver/resolver.cc

```diff
@@ -948,10 +948,75 @@ class ResolveConstantsWalk {
             transformAncestor(ctx.withOwner(klass), singleton, ancst, isInclude);
         }
 
+        // TODO: Only do this check in --stripe-mode. Currently it does the check but skips reporting the error if
+        // not in --stripe-mode.
```

Comment:
Nit: the mentions of `--stripe-mode` in this comment are supposed to be `--stripe-packages`?

Reply:
No, this is different from `--stripe-packages`. We actually want to run it in `--stripe-mode` regardless of `--stripe-packages`.

Reply:
ooc: does it have to be in `--stripe-mode`? i have a feeling this might be something that open sources users might want to turn on more widely. what parts of stripe mode does it actually depend on?

- if it's actually orthogonal and we just want to enable it for stripe's codebase but not necessarily all users' codebases, can we add a separate flag?
- if it depends on some nuanced assumptions that stripe mode sets up, can we attempt to ENFORCE that those hold here and/or document what those assumptions are in a comment?

Reply:
@jez My feeling is that in more predictably built codebases, where filler nodes are created appropriately (as opposed to Stripe where there are some edge cases with our autoloader), this check would not be useful to run?

I could be wrong.

Reply:
Update -- I decided to make this available even outside of `--stripe-mode`.

Reply:
Added a separate flag.